src/plugins/shared_ux/public/components/redirect_app_links/utils.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+
+/**
+ * Returns true if any modifier key is active on the event, false otherwise.
+ */
+export const hasActiveModifierKey = (event: React.MouseEvent): boolean => {
+  return event.metaKey || event.altKey || event.ctrlKey || event.shiftKey;
+};
+
+/**
+ * Returns the closest anchor (`<a>`) element in the element parents (self included) up to the given container (excluded), or undefined if none is found.
+ */
+export const getClosestLink = (
+  element: HTMLElement | null | undefined,
+  container?: HTMLElement
+): HTMLAnchorElement | undefined => {
+  let current = element;
+  do {
```

Comment:
Might not be entirely relevant, but it's something I think of whenever I see a `do/while`:

![](https://pbs.twimg.com/media/DuP34l7XgAApQmq.jpg)

:-D  It will probably make all of this code cleaner, too.

Reply:
I prefer `do...while` over `while(true)` anytime.